### DISCUSSION TITLE
HFEv3 support

### DISCUSSION THREAD

**Discussion by RetroD** [keirf/greaseweazle#468] [Show and tell]
Hello, and first of all thank you Keirf for whar you have created.
I couldn't find any confirmation of this,  but is hfe version3 supported for creating a disk image from a real floppy?

I tried dumping a hfe of an amiga original game directly with gw and tested in my gotek, but the game is not loading.

I could create a scp in gw and then use Hxc to convert the scp to hfe v3, and that read fine on my gotek.

I noticed also a file size difference between the direct hfe dump from gw (about 2.00 MB) compared to the scp to hfev3 export by HxC ( about 2.4 MB)

The game I tested is Mortal Kombat, which has copy protection.

Is gw outputting hfev1, and is there a way to dump a hf3v3 directly?

**Comment by keirf** [comment]
To create an HFEv3 you need to specify version 3 explicitly. For example:
```
gw convert name.scp name.hfe::version=3
```

However, the gw converter is not very good at converting raw dumps to HFEv3, as it gets thrown off by the random noise and generates too many rate-change opcodes in the HFEv3 output. So your mileage may vary! Seems like HxC may be better for conversion of raw dumps. *However* what gw **is** good at is converting clean images (specifically IPF image files) to HFEv3.
```
gw convert name.ipf name.hfe::version=3
```

That said, if you have a dump you can share which is working with HxC but not with gw, I'm happy to take a look and see if I can get well-behaved raw dumps converting for you. I'm sure there are some simple heuristics I could try out.

**Reply by RetroD** [reply]
Thank you for the quick reply!
 I have not tried using the convert command as I wanted to see if I could dump a working HFEv3 directly from an original copy protected floppy. (so I could do the dump in 1 pass rather than dump to scp and then convert)

I have tried this command:
`gw read Floppies\MK_1.hfe::version=3::bitrate=250`

This gave me an error at the end of the dump:
`HFEv3: Track too long to fit in image!
Are you trying to convert raw flux (SCP, KF, etc)?
If so: Try specifying 'uniform': eg. name.hfe::version=3:uniform
       (will break variable-rate copy protections such as Copylock)
If not: Report a bug.`

Next, I tried this:
`read Floppies\MK_1.hfe::version=3:uniform::bitrate=250`

This created an image, slightly larger than before (2.15MB) but it still did not work in the amiga. I think it fails with the copy protection.

Happy to share my SCP if you wanted to do some experiments. Let me know if this link works for you:
https://drive.google.com/file/d/1Nnu72GdesZyM_ixiF_qwHsyNBK1XlUxU/view?usp=drive_link

**Reply by RetroD** [reply]
and this is the HFEv3 that I created in HxC by converting the scp. This one works on the amiga:
https://drive.google.com/file/d/1dlxATDEsolcta8r9OUvFgBWzjIRaT4nv/view?usp=drive_link

**Reply by keirf** [reply]
I will take a look. Indeed the `uniform` option won't work because Mortal Kombat uses Copylock protection which has a variable-rate protection track.

**Comment by RetroD** [comment]
I see, thank you for taking a look.
I wonder if this is what is also causing my inability to successfully write back a scp to floppy, But I will open another thread so it doesn't get mixed up with this.

**Reply by keirf** [reply]
That primarily depends on whether the tracks are all index-aligned. Depends on the disk. Most commercial software is mastered with index-aligned tracks.
